public/app/plugins/panel/heatmap/migrations.ts

```diff
@@ -14,7 +14,7 @@ import { Options, defaultOptions, HeatmapColorMode } from './types';
 /** Called when the version number changes */
 export const heatmapMigrationHandler = (panel: PanelModel): Partial<Options> => {
   // Migrating from angular
-  if (Object.keys(panel.options).length === 0) {
+  if (panel.options === null || Object.keys(panel.options).length === 0) {
```

Comment:
maybe a bit shorter, (and also handles `undefined`):

```suggestion
  if (Object.keys(panel.options ?? {}).length === 0) {
```

Reply:
Thanks, addressed in https://github.com/grafana/grafana/pull/79083/commits/7120465ee0fffbc87bad8d17776f2f1ece6c9387.